gitpod-io/gitpod
# Make the pods be removed when workspacekit fails.

## Description

There was a problem with workspace pendings due to finalizer not working properly when iws failed to start (or had a large delay). This PR avoids that by skipping the finalizer. Also, the workspace phase is now consistent with it.

## Related Issue(s)

Fixes https://github.com/gitpod-io/gitpod/issues/9673

## How to test

- workspaces can normally start. You can confirm [this preview environment](https://to-conn-prob.preview.gitpod-dev.com/workspaces)
- Intentionally break communication between iws and ws-daemon, and when a workspace failed to start, confirm that  the workspace phase is properly stopped(NOTE: in this case, the stopping phase is skipped because we don't need to backup.) You can confirm it on [this preview environment](https://to-failed-ws-start.preview.gitpod-dev.com/workspaces)


## Release Notes

```release-note
ws-manager: Make the pods be removed when workspacekit fails.
```

## Documentation

No
